src/FeatureSet.php

```diff
@@ -121,6 +123,12 @@ public function getTimeProvider()
         return $this->timeProvider;
     }
 
+    public function setTimeProvider(TimeProviderInterface $timeProvider)
+    {
+        $this->timeProvider = $timeProvider;
+        $this->timeGenerator = $this->buildTimeGenerator();
```

Comment:
Yup, due https://github.com/ramsey/uuid/pull/69/files#diff-b8cb7bbf7ebc08e740d40433bd79da3aR169. Proper fix is to instead call `$this->setTimeProvider(new SystemTimeProvider())` in the ctor instead. I probably should explicitely pass the required dependencies to the TimeGeneratorFactory instead of passing the FeatureSet instance, it would make the dependencies more explicit.

I'll update my P/R in a bit.